Add directives reference page (#282)
withastro/docs: src/pages/en/core-concepts/framework-components.md

### Available Hydration Directives

> There are serveral hydration directives available for UI framework components, such as `client:load`, `client:idle`, and more!

Reviewer: ```suggestion
There are serveral hydration directives available for UI framework components: `client:load`, `client:idle`, `client:visible`, `client:media={QUERY}` and `client:only=" "`
```

Can we maybe meet in the middle, and since there are only 5 of them, list them all here??  ;)

> 📚 See our [directives reference](/en/reference/directives-reference#ui-framework-components) page for the full list of hydration directives, and their usage.

Reviewer: ```suggestion
📚 See our [directives reference](/en/reference/directives-reference#ui-framework-components) page for a full description of these hydration directives, and their usage.
```
